jwst/resample/resample_spec.py

```diff
@@ -67,14 +72,169 @@ def __init__(self, input_models, output=None, single=False, blendheaders=False,
 
         # Define output WCS based on all inputs, including a reference WCS
         if resample_utils.is_sky_like(self.input_models[0].meta.wcs.output_frame):
-            self.output_wcs = self.build_interpolated_output_wcs()
+            if self.input_models[0].meta.instrument.name != "NIRSPEC":
+                self.output_wcs = self.build_interpolated_output_wcs()
+            else:
+                self.output_wcs = self.build_nirspec_output_wcs()
```

Comment:
Should these new methods really be confined to just NIRSpec data? The MIRI LRS fixed-slit mode is very similar and can involve nodding/dithering of the source within the slit, just like the case for NIRSpec. Maybe we just need to test it on MIRI LRS to find out.